### PR TITLE
Migrate Channel.GetChannelsByIds to Sync by default

### PR DESCRIPTION
**Pull request by Sheshagiri**
#### Summary
Migrate `Channel.GetChannelsByIds` to Sync by default

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/11161

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -1771,11 +1771,11 @@ func (a *App) AutocompleteChannels(teamId string, term string) (*model.ChannelLi
 
 		channelList := model.ChannelList{}
 		if len(channelIds) > 0 {
-			cresult := <-a.Srv.Store.Channel().GetChannelsByIds(channelIds)
-			if cresult.Err != nil {
-				return nil, cresult.Err
+			channels, err := <-a.Srv.Store.Channel().GetChannelsByIds(channelIds)
```
Comment (jespino) [inline app/channel.go]: You no longer need the `<-` there.

File: app/user.go [hunk 2]
```diff
@@ -2154,11 +2154,11 @@ func (a *App) getListOfAllowedChannelsForTeam(teamId string, viewRestrictions *m
 		return channelIds, nil
 	}
 
-	cresult := <-a.Srv.Store.Channel().GetChannelsByIds(viewRestrictions.Channels)
-	if cresult.Err != nil {
-		return nil, cresult.Err
+	channels, err := <-a.Srv.Store.Channel().GetChannelsByIds(viewRestrictions.Channels)
```
Comment (jespino) [inline app/user.go]: Same here

File: store/storetest/channel_store.go [hunk 3]
```diff
@@ -484,30 +484,28 @@ func testChannelStoreGetChannelsByIds(t *testing.T, ss store.Store) {
 	_, err = ss.Channel().SaveDirectChannel(&o2, &m1, &m2)
 	require.Nil(t, err)
 
-	if r1 := <-ss.Channel().GetChannelsByIds([]string{o1.Id, o2.Id}); r1.Err != nil {
-		t.Fatal(r1.Err)
+	if r1, err := <-ss.Channel().GetChannelsByIds([]string{o1.Id, o2.Id}); err != nil {
```
Comment (jespino) [inline store/storetest/channel_store.go]: unneeded `<-`

File: store/storetest/channel_store.go [hunk 4]
```diff
@@ -484,30 +484,28 @@ func testChannelStoreGetChannelsByIds(t *testing.T, ss store.Store) {
...
-		if cl[1].ToJson() != o2.ToJson() {
+		if r1[1].ToJson() != o2.ToJson() {
 			t.Fatal("invalid returned channel")
 		}
 	}
 
 	nonexistentId := "abcd1234"
-	if r2 := <-ss.Channel().GetChannelsByIds([]string{o1.Id, nonexistentId}); r2.Err != nil {
-		t.Fatal(r2.Err)
+	if r2, err := <-ss.Channel().GetChannelsByIds([]string{o1.Id, nonexistentId}); err != nil {
```
Comment (jespino) [inline store/storetest/channel_store.go]: unneeded `<-`

**Comment by Sheshagiri** [comment]
> Just a repeated syntax problem, but the rest of the code looks good to me. Thanks @Sheshagiri! 🎉

I thought I removed that. Fixed it now.

**Pull request merged**